src/EditorFeatures/Core.Wpf/Options/LegacyEditorConfigDocumentOptionsProvider.DocumentOptions.cs

```diff
@@ -40,10 +41,10 @@ public bool TryGetDocumentOption(OptionKey option, OptionSet underlyingOptions,
                 // Dictionary<string, object> from the old system.
                 // 
                 // We cache this with a conditional weak table so we're able to maintain the assumptions in EditorConfigNamingStyleParser
-                // that the instance doesn't regularly change and thus can be used for further caching
+                // that the instance doesn't regularly change and thus can be used for further caching.
                 var allRawConventions = s_convertedDictionaryCache.GetValue(
                     _codingConventionSnapshot.AllRawConventions,
-                    d => ImmutableDictionary.CreateRange(d.Select(c => KeyValuePairUtil.Create(c.Key, c.Value.ToString()))));
+                    d => new StringConvertingDictionary(d));
```

Comment:
i don't really like this.  I don't see anything in dictionary that guarantees order.  It seems like if we care about that... we should use a specific dictionary impl that guarantees taht, right?

Note: i can't find good docs here, but hte main docs say: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.system-collections-idictionary-keys?view=netframework-4.8#remarks

> The order of the keys in the returned ICollection is unspecified, but it is guaranteed to be the same order as the corresponding values in the ICollection returned by the Values property.

Reply:
FWIW, we don't like it either 😆

Reply:
:D

Reply:
Yes, we totally regret this, but time is not on our side both in that we don't have a time machine to easily fix this, and also we don't have time before 16.1 ships to redesign this.

Reply:
understood.  i get that using an IReadOnlyDictionary might be something you can't control. However, at least in the future, it might be good ot use an explicit implementation of IReadOnlyDictionary that we control which explicitly verifies enumeratino order.  Not a huge biggie, just something to consider :)